bpf/lib/nodeport.h

```diff
@@ -1640,14 +1644,22 @@ int tail_nodeport_nat_ingress_ipv4(struct __ctx_buff *ctx)
 
 	/* At this point we know that a reverse SNAT mapping exists.
 	 * Otherwise, we would have tail-called back to
-	 * CALL_IPV4_FROM_NETDEV in the code above. The existence of the
-	 * mapping is an indicator that the packet might be a reply from
-	 * a remote backend. So handle the service reverse DNAT (if
-	 * needed)
+	 * CALL_IPV4_FROM_NETDEV in the code above.
+	 */
+#if !defined(ENABLE_DSR) || (defined(ENABLE_DSR) && defined(ENABLE_DSR_HYBRID)) ||	\
```

Comment:
I think it would be worth having a code comment here to mention why we have a condition on DSR all of a sudden. IIUC, something like "Reverse DNAT for DSR was performed at the backend's node so we can skip that case for the intermediate node here."

Reply:
Added a comment

> If we're not in full DSR mode, reply traffic from remote backends might pass back through the LB node and requires revDNAT.